nemo/collections/nlp/data/language_modeling/megatron/blendable_dataset.py

```diff
@@ -74,7 +74,12 @@ def __len__(self):
     def __getitem__(self, idx):
         dataset_idx = self.dataset_index[idx]
         sample_idx = self.dataset_sample_index[idx]
-        return self.datasets[dataset_idx][sample_idx]
+        # Ensure the sample index doesn't exceed the dataset size
+        # original build_index function does not handle the extreme case properly
+        sample_idx = sample_idx % len(self.datasets[dataset_idx])
```

Comment:
Add a warning here for out of boundary idx and mentioning it's reusing existing examples